### ISSUE TITLE
Feature request: add arbitrary files to jars

### ISSUE THREAD

**Issue by cuchaz** [jerkar/jerkar#37]
Properly adding license/readme files to jars is tricky because license/readme files belong in the project root, not in any source/resources folder. So they don't get "compiled" into the classes folder and picked up by the usual jar packer methods.

Here's some workaround code I have that's working for now:
```
	@Override
	protected JkJavaPacker createPacker() {
		return JkJavaPacker.builder(this)
			.includeVersion(true)
			.doJar(false)
			.doSources(false)
			.doFatJar(true)
			.extraAction(new JkExtraPacking() {

				private File licenseFile = new File("LICENSE.txt");
				
				@Override
				public void process(JkJavaBuild build) {
					
					// move the fat jar to a temp file
					JkJavaPacker packer = createPacker();
					File tempFile = build.ouputDir("temp.jar");
					packer.fatJarFile().renameTo(tempFile);
					
					// make a new fat jar with the stuff we want to keep
					try (ZipFile zipIn = new ZipFile(tempFile)) {
						try (ZipOutputStream zipOut = JkUtilsZip.createZipOutputStream(packer.fatJarFile(), Deflater.DEFAULT_COMPRESSION)) {
							
							// add new entries
							JkUtilsZip.addZipEntry(zipOut, licenseFile, licenseFile.getName(), false);
						}
					} catch (IOException ex) {
						throw new RuntimeException(ex);
					}
				
					// cleanup
					tempFile.delete();
				}
			})
			.build();
	}
```

Maybe there's a way to tell the default packer to grab extra files while it's zipping the classes folder?

Thanks!

**Comment by djeang** [comment]
I think something like this can make it 
```
@Override
public JkFileTreeSet editedResources() {
        return super.editedResources().and(new File("../myRootProject/LICENSE.TXT"));
}
```

**Comment by cuchaz** [comment]
that doesn't seem to work =(
`java.lang.IllegalArgumentException: LICENSE.txt is not a directory.`

**Comment by djeang** [comment]
Indeed. Sorry I put you on the wrong track. Resources are managed as file tree so you can decide their relative path to the root. This should work 

```
@Override
public JkFileTreeSet editedResources() {
        JkFileTree licenceFiles = baseDir().from("../myRootProject")
                .andFilter(JkPathFilter.include("license.txt"));    
        return super.editedResources().and(licenceFiles);
}
    
```

**Comment by cuchaz** [comment]
That worked, thanks! Your solution is much better than mine. =P

Maybe a mention of this trick in the documentation would be helpful to new jerkar users. It's not obvious to new users that the editedXXX() methods can use full file trees/forests and not just a single folder.

**Comment by djeang** [comment]
Ok, did you find it by yourself if the method was named xith other than 'editedResources' ? I am trying to find the most intuitive way to do it ....

**Comment by cuchaz** [comment]
I think the code is fine, but the documentation should just show a few examples of common use cases for overriding editedXXX() methods so users can learn how to do them. Adding README.txt and LICENSE.txt files to a jar or fat jar would be a good example. Or any other case where you want to add a file to a jar file for humans to read, rather than the JVM.

**Comment by cuchaz** [comment]
implemented by #45
